miner/worker.go

```diff
@@ -711,82 +767,60 @@ func (w *worker) commitUncle(env *environment, uncle *types.Header) error {
 	if env.family.Contains(hash) {
 		return errors.New("uncle already included")
 	}
-	env.uncles.Add(uncle.Hash())
+	env.uncleHashes.Add(uncle.Hash())
+	env.uncles = append(env.uncles, uncle)
```

Comment:
how about using a `map[common.Hash]*types.Header` here?

Reply:
Yes, why not